### DISCUSSION TITLE
Support for CockroachDB for scalability & SQLite for simplicity

### DISCUSSION THREAD

**Discussion by deshetti** [umami-software/umami#981] [Ideas]
Hi @mikecao 

I have read through the docs, your medium post, issues, and discussions to understand that the target group for Umami in your opinion is folks with one or a few websites with limited traffic.

Do you see any limitations in why the solution would not scale to hundreds, if not thousands, and more websites with millions of hits on one instance (deployed on kubernetes)? I would think Database would be the biggest sticking point. I see that you are using Prisma as ORM and it does support CockroachDB which is horizontally scalable. I don't have any experience with Prisma, but I would think as an ORM the same code (maybe with minimal changes) should support CockroachDB as well. Is there a reason apart from the work/testing/support that would limit the solution for supporting CockroachDB?

If Database scalability is taken care of, I would think scaling the rest of the application itself on Kubernetes is pretty straightfoward. I have created a Helm chart to test this and it works pretty well so far if anyone wants to install it and give it a shot: https://github.com/factly/helm-charts/tree/main/charts/umami

On the flip side, I would argue that for folks who are looking for a lightweight solution - Postgres or MySQL could still be avoided if Umami can support SQLite (which Prisma supports as well).

**Comment by mikecao** [comment]
For very simple queries, Prisma is usable across different databases. However, the complex queries require different date functions that are specific to certain databases. That's why there's no SQLite support even though Prisma supports it. I just haven't found the time to dive in and figure out those queries for SQLite.
